app/models/user.rb

```diff
@@ -230,8 +230,28 @@ def mfa_api_authorized?(otp)
   end
 
   def mfa_required?
-    return false if mfa_enabled?
-    owner_of_most_downloaded_gem?
+    return false unless require_mfa_at
+    require_mfa_at < Time.now.utc 
+  end
+
+  def recommend_mfa_setup?
+    return false unless require_mfa_at && !mfa_enabled?
+    require_mfa_at > Time.now.utc 
+  end
```

Comment:
Discussed with Jenny, we'll be retiring this implementation in favour of `#recommend_setup_for_mfa?`. 👍